docs/other_resources/high_level_architecture.md

```diff
@@ -1,3 +1,8 @@
 # High level architecture
 
+### Component overview
 <img src="../images/of-flagd-0.png" width="560">
+
+### Flag configuration provider
```

Comment:
Should these be H2s? Also, could you add a brief description under each section?

Reply:
Yeah they can be H2. I will update the sections with next commit :)